Added 'Events' table

DishantK1807 opened the pull request:
Added 'Events' to the update.sql. #82 
Removed Whitespaces and added comments

DishantK1807 committed:
Added 'Events' table

hongquan commented on db/update.sql:
```diff
@@ -1,6 +1,6 @@
 -- drivers license information
 CREATE TABLE IF NOT EXISTS `UserDriverLicenses` (
-  `user_id` int(11) NOT NULL,
+  `user_id` int(11) NOT NULL AUTO_INCREMENT,
```
Why do you make this field "AUTO_INCREMENT"?

Note below:

```
ADD CONSTRAINT `userdriverlicenses_ibfk_1` FOREIGN KEY (`user_id`) REFERENCES `User` (`UID`) ON DELETE CASCADE ON UPDATE CASCADE;
```

this field is a **foreign key**

DishantK1807 commented:
@hongquan : Will update that.

hongquan commented on db/update.sql:
```diff
@@ -27,3 +28,15 @@ ALTER TABLE `User`
+  `description` text NOT NULL,
+  `organizer` varchar(255) NOT NULL,
+  `event_date` int(11) NOT NULL,
+  `event_time` int(11) NOT NULL,
```
What are the two fields `event_date`, `event_time` for? Is it starting date or ending date of event?

How do we store date, time when you choose to store as integer type? MySQL has [DATETIME](http://dev.mysql.com/doc/refman/5.5/en/datetime.html) type for storing date, time data.

DishantK1807 replied:
@hongquan : `event_date` is for date on which the event is and `event_time` is the starting time of the event. 
In all the other tables date and time are stored in integer type only. I was trying to follow that convention only.

Table Shift from install.sql:

`DROP TABLE IF EXISTS`Shifts`;
CREATE TABLE IF NOT EXISTS`Shifts`(
`SID`int(11) NOT NULL AUTO_INCREMENT,
`title`text,
`shifttype_id`int(11) NOT NULL,
`start`int(11) NOT NULL,
`end`int(11) NOT NULL,
`RID`int(11) NOT NULL DEFAULT '0',
`URL`text,
`PSID`int(11) DEFAULT NULL,
`created_by_user_id`int(11) DEFAULT NULL,
`created_at_timestamp`int(11) NOT NULL,
`edited_by_user_id`int(11) DEFAULT NULL,
`edited_at_timestamp`int(11) NOT NULL,
  PRIMARY KEY (`SID`),
  UNIQUE KEY`PSID`(`PSID`),
  KEY`RID`(`RID`),
  KEY`shifttype_id`(`shifttype_id`),
  KEY`created_by_user_id`(`created_by_user_id`),
  KEY`edited_by_user_id`(`edited_by_user_id`)
) ENGINE=InnoDB  DEFAULT CHARSET=utf8 AUTO_INCREMENT=133 ;
`

Should I change it to `DATETIME` type? or follow the existing pattern?

DishantK1807 committed:
removed AUTO_INCREMENT from 'user_id'

hongquan commented:
@DishantK1807 How about if the event span on several day? Which day the `event_date` variable represent?

You can follow the existing pattern (use integer to store).

DishantK1807 commented:
@hongquan: `event_date` represents the date on which the event is. For the events with span of several days I'll add start and end date to the db.

DishantK1807 committed:
added end_date 

Added `end_date` for the events with span of several dates. `event_date` will be used as the start date as there is no need to maintain redundant information

DishantK1807 commented:
@hongquan : Can we merge this PR now to start with next steps or we need to add more information to the table?

hongquan commented:
The name `event_date` is confusing. If you have `end_date`, why don't you rename `event_date` to `start_date` to make them more relate?

I think again, we should use MySQL's DATETIME type, it will help when you need to query database based on time, for example, `SELECT` or `COUNT` something in a time range. Also, when you fetch data to PHP, those data in DATETIME will be converted automatically to PHP's `DateTime` class.

So, `start_date` will be the date and time the event starts, and `end_date` will be the date and time the event will end.

You can merge _time_ & _date_ columns to one, because DATETIME already include time.

DishantK1807 commented:
@hongquan : Would we be requiring separate field for event start time, `event_time`,  as well or should I remove it as we will be merging date and time?

hongquan commented:
> So, start_date will be the date and time the event starts, and end_date will be the date and time the event will end.

> You can merge time & date columns to one, because DATETIME already include time.

Of course after you merge the two columns, one unused column has to be removed.

DishantK1807 committed:
changed int to datetime

DishantK1807 commented:
@hongquan : Done!

hongquan commented:
Please use the lowest supported date for the default value http://dev.mysql.com/doc/refman/5.6/en/datetime.html

DishantK1807 commented:
@hongquan: Do we convert the datatype for the existing fields as well?
And should I set the year to `2000-00-00` for the default value?

DishantK1807 commented:
Also it will be easier to check for invalid dates with the current values.

hongquan commented:
@DishantK1807 You can update the existing fields if you can manage time to do (not high priority).
Should be '2000-01-01' (day 0 month 0 is not valid)

DishantK1807 committed:
Updated default values

DishantK1807 commented:
@hongquan : Will look into it.

DishantK1807 commented:
Done!
